numpy/random/mtrand/mtrand.pyx

```diff
@@ -4888,6 +4889,247 @@ cdef class RandomState:
+        The name ``colors`` comes from a common description of the
+        distribution: it is the probability distribution of the number of
+        marbles of each color selected without replacement from an urn
+        containing marbles of different colors; ``colors[i]`` is the number
+        of marbles in the urn with color ``i``.
+
+        Parameters
+        ----------
+        colors : sequence of integers
+            The number of each type of item in the collection from which
+            a sample is drawn.  The values in ``colors`` must be nonnegative.
+        nsample : int
+            The number of items selected.  ``nsample`` must not be greater
+            than ``sum(colors)``.
+        size : int or tuple of ints, optional
+            Output shape.  If the given shape is, e.g., ``(m, n, k)``,
+            then ``m * n * k`` samples are drawn, and the return value has
+            shape ``(m, n, k, len(colors))``.  If `size` is an integer,
+            the output has shape ``(size, len(colors))``.  Default is None,
+            in which case a single sample is returned as an array with shape
+            ``(len(colors),)``.
+
+        Returns
+        -------
+        sample : ndarray
+            Sample drawn from the multivariate hypergeometric distribution.
+
+        See Also
+        --------
+        hypergeometric : Draw samples from the (univariate) hypergeometric
+            distribution.
+
+        Notes
+        -----
+        .. versionadded:: 1.12.0
+
+        Examples
+        --------
+        >>> colors = [16, 8, 4]
+        >>> np.random.seed(12345)
+        >>> np.random.multivariate_hypergeometric(colors, 6)
+        array([3, 1, 2])
+        >>> np.random.multivariate_hypergeometric(colors, 6, size=3)
+        array([[3, 1, 2],
+               [2, 2, 2],
+               [3, 3, 0]])
+        >>> np.random.multivariate_hypergeometric(colors, 6, size=(2, 2))
+        array([[[2, 4, 0],
+                [4, 1, 1]],
+
+               [[4, 2, 0],
+                [3, 2, 1]]])
+
+        """
+        colors_arr = <ndarray> PyArray_ContiguousFromObject(colors, NPY_LONG,
+                                                            0, 0)
+        if colors_arr.ndim != 1:
+            raise ValueError("colors must be a one-dimensional sequence "
+                             "of integers.")
+        if np.any(colors_arr < 0):
+            raise ValueError("values in colors must be nonnegative.")
+
+        if nsample < 0:
+            raise ValueError("nsample must be nonnegative.")
+
+        ncolors = len(colors)
+        total = np.sum(colors)
+        if nsample > total:
+            raise ValueError("nsample > sum(colors)")
+
+        if ncolors == 0:
+            if size is None:
+                return np.array([], dtype=int)
+            else:
+                return np.empty(size + (0,), dtype=int)
+
+        # Heuristic rule for choosing the algorithm.
+        if total > 500000 or nsample > 12*ncolors:
```

Comment:
What is the reason for this choice? Note that if it is changed the output sequence may also be changed.